kratos/processes/find_global_nodal_elemental_neighbours_process.cpp

```diff
@@ -126,33 +122,23 @@ namespace Kratos
                 [](GlobalPointer<Element> const& gp){return gp->Id();}
         );
 
-        #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(mr_model_part.Nodes().size()); ++i)
-        {
-            auto it = mr_model_part.NodesBegin() + i;
-            auto& neighbours = it->GetValue(NEIGHBOUR_ELEMENTS);
+        block_for_each(mr_model_part.Nodes(), [&](Node<3>& rNode){
+            auto& neighbours = rNode.GetValue(NEIGHBOUR_ELEMENTS);
             neighbours.shrink_to_fit();
             std::sort(neighbours.ptr_begin(), neighbours.ptr_end(),
                 [&id_proxy](GlobalPointer<Element> const& gp1, GlobalPointer<Element> const& gp2)
                 {
                     return id_proxy.Get(gp1) < id_proxy.Get(gp2);
                 }
             );
-        }
-    
+        });
     }
 
     void FindGlobalNodalElementalNeighboursProcess::ClearNeighbours()
     {
         NodesContainerType& rNodes = mr_model_part.Nodes();
 
-        #pragma omp parallel for
-        for(int i=0; i<static_cast<int>(rNodes.size()); ++i)
-        {
-            auto in = rNodes.begin() + i;
-            auto& rN = in->GetValue(NEIGHBOUR_ELEMENTS);
-            rN = GlobalPointersVector< Element >();
-        }
+        VariableUtils().SetVariable(NEIGHBOUR_ELEMENTS, GlobalPointersVector< Element >(), rNodes);
```

Comment:
@philbucher I guess this will be correct.

Reply:
yes